### ISSUE TITLE
[ZE] Sistema de punição ingame por rounds

### ISSUE THREAD

**Issue by danielhidalgojunior** [quasemago/zrageservers#110]
Alinhei umas ideias com o Gabi e acho que seria interessante um sistema de punição ingame similar ao que existe hoje no TTT. Seria aplicado em casos de infrações **leves e medianas**, onde um aviso/kick é muito leve, um ban, mesmo que curto, é muito pesado e quando a infração não é relacionada a mal uso de matérias.

Obviamente, não podemos dar slay no punido todo round pois acabaria prejudicando o humans com a quantidade de zombies. Talvez o punido pudesse ser transformado junto com os mothers mas com uma **classe de zombie bem batata**, sem poder de infecção, mais lento, maior knockback, com bem menos vida e com uma skin específica para que os humanos não se assustem. 

E também aplicar um **eban automático** junto a punição para evitar que o cara pegue alguma matéria antes da infecção pra tentar jogar longe ou coisas do tipo.

Seria bom se o punido também tivesse um **hud com uma mensagem bem grande na tela mostrando os rounds punido, motivo e como o punish afeta o gameplay dele (não pode infectar, não pode pegar materia, mais lento, etc)**. Isso ajudaria os novatos a entender o que aconteceu e tbm pra diminuir a comunicação desnecessária entre admin e o punido (explicar que não vai conseguir infectar, vai jogar de human dnv só daqui tantos rounds bla bla bla).

O que acham?

**Comment by alminha021** [comment]
A principio nao gostei. 

Mas se for pra frente: 

Não atrapalhando a infecção padrão, da pra pensar.
- O punido NAO pode ser considerado mother senao for a vez dele, ou seja:
se 6 pessoas serão escolhidas como mother , que seja 6 pessoa normamelte + os punidos. (por exemplo: se forem 3 punidos , seriam 9 zm , 6 mother + 3 punidos)
BUT: se algum punido estiver na frente da fila e for virar mother, que ele ainda possa infectar , ams seja mais lento e com kb maior.

- Hp dele tem q ser baixo e o kb bem mais alto, mas acho importante um freeze automatico de uns 5s neles no começo, existem materias de zm que eles podem acabar bugando, e existem mapas que se zms ativarem triggers, os ct perdem, logo ele ainda eh uma ameaça , tendo que mata-lo tbm e segura-lo.

Esse sistema tem que funcionar de uma maneira que nao facilite ainda mais a vida dos ct's e prejudique o time de zm. Por isso acho importante que nao atrapalhe a fila de infecção normal, caso ele seja mother que ainda possa infectar.

**Comment by lucaslinha** [comment]
Deixa o punido mais lento que o ct, kbk alto, napalm alto, eban enquanto durar e não pode ganhar ponto de zm tmb.

**Comment by danielhidalgojunior** [comment]
creio que da maneira atual, o punido não irá atrapalhar nenhum time visto que ele não pode infectar e tbm n é humano para segurar zombies.

Sobre a fila, creio que seria melhor pular o punido e dar o mother para outro jogador.

Essa questão de ativar triggers seria uma vantagem pra zm, mas com slow e vida baixa acho que ele não teria muitas chances.

**Comment by hcjordao** [comment]
Acho interessante mas muito tramposo. Ao meu ver um eban é suficiente no cara e se tiver trolando um kick e depois ban. Jogo de TTT é rapido pros punishes mesmo sendo 4 rounds passem em menos de 6 minutos. Isso dai demoraria um round de ZE inteiro para passar que pode ser algo bem curto até 20 minutos de round.

**Comment by danielhidalgojunior** [comment]
bom, a quantidade de rounds deve variar de acordo com o tempo de cada round do mapa

**Comment by lucaslinha** [comment]
> Acho interessante mas muito tramposo. Ao meu ver um eban é suficiente no cara e se tiver trolando um kick e depois ban. Jogo de TTT é rapido pros punishes mesmo sendo 4 rounds passem em menos de 6 minutos. Isso dai demoraria um round de ZE inteiro para passar que pode ser algo bem curto até 20 minutos de round.

Odeio kick pq é uma punição que não é justa o suficiente, alem de que o server perde menos 1 player jogando, afinal a maioria que toma kick não volta mais, prefiro que o cara continue jogando e dando ibope ao sv ao mesmo tempo que paga por seus pecados. 
Sobre rounds, basta apenas usar tempo.

**Comment by hcjordao** [comment]
Concordo nesse ponto que tu disse de dar ibope ao server. Botar na prática então para testar. Só precisa definir o que é necessário ser feito.

**Comment by danielhidalgojunior** [comment]
punir por tempo tem seus contras, pq a punição é vc ver o time ganhando e vc ser um zm batata que não pode fazer nada, daí se o cara ta punido por 15 min mas o round dura 20 min, ele será punido praticamente uma vez só; Em contrapartida, rounds warmup ou batatisse do CTs morrendo mt rápido no começo do mapa podem amenizar e muito a punição dos caras caso essa seja por rounds.

Talvez permitir ambos métodos? Ex: sm_punish haru 2r "troll" (2 rounds), sm_punish haru 20 "troll" (20 minutos)

**Comment by gabrielacms** [comment]
Eu acho melhor ser só um tipo de punição. Ou é por round ou é por tempo. Por questão de facilidade mesmo na hora de executar o comando. Não me importo do cara perder a mother, eu gostei da ideia, vejo ela sendo aplicada só em caso de ter ficado na ponta e matou o resto do time.  Só que prefiro sendo por round do que por tempo. Funcionando da mesma maneira do TTT, tomou punish de 2 rounds, então vai ser 2 rounds seguidos de zm. 
- Zombie + lento -> concordo
- Sem ponto por ganhar como zm -> concordo.

**Comment by alminha021** [comment]
entao como ficaria?

 zombie:
- +lento - acho q todos concordam
- +kb alto - tbm?
-  poder de infecçao retirado - eu sou a favor dependendo de como for feito , se ele vai ser mother nos rounds q tomar slay ou nao.
- nao ganhar pontos de zm - concordo tbm


comando:
- por rounds eh melhor ( tem q ver como que fica o eban , se tem como fazer , colocar como permanente qnd tomar punish e retirar qnd acabar a puniçao, acho q resolve)
- utiliza-lo apos alertar o infrator um avez e ele repetir? ou como no ttt que o cara nao le regra e faz besteira e ja eh aplicada a puniçao?
 acho que aqui vai do bom senso do admin

**Comment by danielhidalgojunior** [comment]
então, tem que levar várias coisas em conta na hora de punir. É macaco velho? Joga há quanto tempo? Foi intencional? houveram prejudicados? Creio que dá pra substituir muito kick e aviso por 1 round de punição, por exemplo. No caso de report, creio que na maioria dos casos poderia dar o aviso e em seguida a punição, mesmo que seja a mais leve possível.

O problema vai ser definir regras de conduta pra essas punições visto que a punição deve variar de acordo com o mapa. Talvez usar um cálculo com base na dificuldade em estrelas do mapa?

**Comment by gabrielacms** [comment]
Ainda há a necessidade de implantação desse tipo de punição? Hoje em dia, sinceramente acho que não precisa.
